cmd/compute-domain-controller/computedomain.go

```diff
@@ -297,6 +330,27 @@ func (m *ComputeDomainManager) onAddOrUpdate(ctx context.Context, obj any) error
 		return fmt.Errorf("error adding finalizer: %w", err)
 	}
 
+	// A sufficiently determined user may have created and then deleted workload
+	// in fast succession so that so that some CD node labels may have been
+	// applied _after_ RemoveNodeLabels() was called for that CD (as part of CD
+	// deletion). That can result in dangling CD node labels, blocking affected
+	// nodes from being re-used for another CD until dangling labels get
+	// removed. In order to keep formation of a new CD fast: do not wait for the
+	// next periodic cleanup to happen, but trigger removal of dangling
+	// ComputeDomain node labels now. Run asynchronously for further latency
+	// reduction (give up control: don't need to explicitly wait for this to
+	// complete). Throttle the frequency of that (upon controller start or when
+	// CDs get created at high-ish frequency then this manual cleanup is invoked
+	// at unnecessarily high frequency): queue up at most 1 cleanup, with a
+	// counting semaphore / fixed-size queue.
+	select {
+	case m.cleanupManager.sem <- struct{}{}:
+		// Counting semaphore acquired
+		go m.cleanupManager.cleanup(ctx)
+	default:
+		// Could not acquire semaphore
+	}
+
```

Comment:
I think this can / should be inside the cleanup() call rather than out here.

Reply:
Agree. Now proposing
```
// EnqueueCleanup() submits a cleanup task if the queue is currently empty.
// Return a Boolean indicating whether the task was submitted or not.
```